dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIHealthCheck.java

```diff
@@ -1,23 +1,40 @@
 package io.dropwizard.jdbi;
 
 import com.codahale.metrics.health.HealthCheck;
+import io.dropwizard.util.Duration;
 import org.skife.jdbi.v2.DBI;
 import org.skife.jdbi.v2.Handle;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
 
 public class DBIHealthCheck extends HealthCheck {
     private final DBI dbi;
     private final String validationQuery;
-
-    public DBIHealthCheck(DBI dbi, String validationQuery) {
+    private Duration duration;
+    private ExecutorService executorService;
+    
+    public DBIHealthCheck(ExecutorService executorService, DBI dbi, String validationQuery, Duration duration) {
         this.dbi = dbi;
         this.validationQuery = validationQuery;
+        this.duration = duration;
+        this.executorService = executorService;
     }
 
     @Override
     protected Result check() throws Exception {
+        Result result;
         try (Handle handle = dbi.open()) {
```

Comment:
I support this. 
Connection opening should be perfomed in the background thread. It protect us from network failures.